packages/veui/src/mixins/searchable.js

```diff
@@ -68,17 +68,11 @@ function search (
       parts: [],
       matched: false
     }
-    let offsets = false
-    try {
-      offsets = matchFn(item.item, keyword, options)
-    } catch (e) {
-      console.warn(`error on calling match: ${e}`)
-    }
+    let offsets = matchFn(item.item, keyword, options)
     let isArray = Array.isArray(offsets)
-    let isBool = typeof offsets === 'boolean'
+    let isBool = !isArray && typeof offsets === 'boolean'
     if (!isArray && !isBool) {
-      console.warn(`invalid result of match: ${offsets}`)
-      offsets = false
+      throw new Error(`invalid result of match: ${offsets}`)
```

Comment:
``'The return value of the `match` function must either be a boolean or an array.'``